app/components/fixup_embeds.py

```diff
@@ -20,26 +20,28 @@
 
     from app.bot import GhosttyBot
 
-
-# Ignore `)` if at the end of the link
+VALID_HTTP_CHARS = r"[A-Za-z0-9-._~:/?#\[\]@!$&'()*+,;%=]"
```

Comment:
```suggestion
VALID_URL_CHARS = r"[A-Za-z0-9-._~:/?#\[\]@!$&'()*+,;%=]"
```
This has nothing to do with the protocol AFAIK.